src/azure-cli/azure/cli/command_modules/appconfig/_params.py

```diff
@@ -89,12 +94,26 @@ def load_arguments(self, _):
         c.argument('enable_public_network', options_list=['--enable-public-network', '-e'], arg_type=get_three_state_flag(), is_preview=True,
                    help='When true, requests coming from public networks have permission to access this store while private endpoint is enabled. When false, only requests made through Private Links can reach this store.')
         c.argument('disable_local_auth', arg_type=get_three_state_flag(), is_preview=True, help='Disable all authentication methods other than AAD authentication.')
+        c.argument('retention_days', arg_type=retention_days_arg_type, is_preview=True)
+        c.argument('enable_purge_protection', options_list=['--enable-purge-protection', '-p'], arg_type=get_three_state_flag(), is_preview=True, help='Property specifying whether protection against purge is enabled for this App Configuration. Setting this property to true activates protection against purge for this App Configuration and its contents. Enabling this functionality is irreversible.')
```

Comment:
Could we abstract `enable_purge_protection` into `CLIArgumentType`, so that both `appconfig create` and `appconfig update` commands can reuse it?

Reply:
We are using `get_three_state_flag()` for this argument. When moved to `CLIArgumentType`, somehow the arguments are now being passed as a `str` instead of `bool`. This is an added overhead to convert the `str` to `bool` each time.
Also, if someone forgets to convert it to `bool` it would land up in unexplainable errors.
Do you know any workaround for this?

Reply:
Does this problem only occur when it is defined as `CLIArgumentType`? 
If so, do you add the `arg_type=get_three_state_flag()` in the `CLIArgumentType`? Could you show me the code after you abstract it into `CLIArgumentType`?

Reply:
```python
enable_purge_protection_arg_type = CLIArgumentType(
        options_list=['--enable-purge-protection', '-p'],
        arg_type=get_three_state_flag(),
        is_preview=True,
        help='Property specifying whether protection against purge is enabled for this App Configuration. Setting this property to true activates protection against purge for this App Configuration and its contents. Enabling this functionality is irreversible.'
    )
```
and usage - 
```python
c.argument('enable_purge_protection', arg_type=enable_purge_protection_arg_type)
```
This returns `str` instead of `bool`.

If you put `type=bool` in the `CLIArgumentType`, it results in a compiler error.

Reply:
The problem is that `is_preview=True` cannot be supported in `CLIArgumentType` at present, which will affect the resolution of `arg_type=get_three_state_flag()`. So please keep the original design~